ddl/ddl_api.go

```diff
@@ -2231,6 +2267,10 @@ func (d *ddl) DropColumn(ctx sessionctx.Context, ti ast.Ident, colName model.CIS
 	}
 
 	err = d.doDDLJob(ctx, job)
+	// column not exists, but if_exists flags is true, so we ignore this error.
+	if ErrCantDropFieldOrKey.Equal(err) && spec.IfExists {
```

Comment:
Other kinds of errors are ignored?!

Reply:
? you mean to add a special error to represent _column not exists_ rather than `ErrCantDropFieldOrKey`? but
> ErrCantDropFieldOrKey returns for dropping a non-existent field or key.

Reply:
`err = d.doDDLJob(ctx, job)`  
The `err` could be any kinds of errors, maybe a network error, tikv temporary unavailable...  did you handle those cases?

Reply:
`if ErrCantDropFieldOrKey.Equal(err)`

Reply:
I see